sympy/core/mul.py

```diff
@@ -866,6 +866,8 @@ def _eval_expand_mul(self, **hints):
         # to 1/x*1/(x + 1)
         expr = self
         n, d = fraction(expr)
+        if n/d == n:
+            expr = n/d
```

Comment:
Why do we need to check `if n/d == n`?

Reply:
Hi Oscar. My first pull request only added that second line, `expr = n/d`. However, 2 tests failed when I submitted it, so I assumed I wasn't being specific enough. Since this is only supposed to fix `gcd` when a single unevaluated `Mul` is involved (with no denominator), I added the check `n/d == n` so that it would only redefine `expr` if was safe to ignore the denominator (and thus equal to 1). Not sure if there's a simpler way to do it, but the tests passed with this change so I just left it.

Reply:
It might be that the tests that failed have a slightly changed form of output but are still giving correct results. We should investigate that.

Limiting this to the case where `n/d==n` makes this less useful because there will be other cases where the fix is needed that don't pass that check e.g. something like `Mul(2, 3, 1/x, evaluate=False)`.

Reply:
Makes sense. How should I go about investigating this? Should I revert the changes and let the tests run again? Or will the failed test results which were included in the pull request comments do? I should add that I am not very familiar with the code structure yet so I'd love any help debugging this.